docs/core/event_handler/bedrock_agents.md

```diff
@@ -0,0 +1,291 @@
+
+### Generating OpenAPI schemas
+
+Use the `get_openapi_json_schema` function provided by the resolver.
+This function will produce a JSON-serialized string that represents your OpenAPI schema.
+You can print this string or save it to a file. You'll use the file later when creating the Agent.
+
+=== "Generating the OpenAPI schema"
+
+    ```python hl_lines="24 25"
+    --8<-- "examples/event_handler_bedrock_agents/src/generating_openapi_schema.py"
+    ```
+
+	1. This ensures that it's only executed when running the file directly, and not when running on the Lambda runtime.
+    2. You can use [additional options](#customizing-openapi-metadata) to customize the OpenAPI schema.
+
+=== "OpenAPI schema"
+
+    ```json hl_lines="13 16 24"
+    --8<-- "examples/event_handler_bedrock_agents/src/generating_openapi_schema.json"
+    ```
+
+To get the OpenAPI schema, run the Python script from your terminal.
+The script will generate the schema directly to standard output, which you can redirect to a file.
+
+```sh
+python app.py > schema.json
+```
+
+### Crafting effective OpenAPI schemas
+
+Working with Agents for Amazon Bedrock will introduce [non-deterministic behaviour to your system](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-how.html#agents-rt){target="_blank"}.
+The OpenAPI schema provides context and semantics to the Agent that will support the decision process for invoking our Lambda function. Sparse or ambiguous schema can result in unexpected outcomes.
+
+We recommend enriching your OpenAPI schema with as many details as possible, which will help the Agent understand your functions and make correct invocations. To achieve that:
+
+* always describe your function behaviour using the `description` field in your annotations
+* in case of refactoring, update your description field to match the function outcomes
+* use distinct `description` for each function to have clear separation of semantics
+
+### Walkthrough for creating the agent
+
+To create an Agent for Amazon Bedrock, refer to the [official documentation](https://docs.aws.amazon.com/bedrock/latest/userguide/agents-create.html) provided by AWS.
+
+The following video demonstrates the end-to-end process:
+
+<center>
+<iframe width="720" height="405" src="https://www.youtube-nocookie.com/embed/NWoC5FTSt7s?si=AG2qpLJbxCkyiLma&amp;controls=1" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowfullscreen></iframe>
+</center>
+
+During the creation process, you should use the schema generated in the previous step when prompted for an OpenAPI specification.
+
+## Advanced
+
+### Accessing custom request fields
+
+The event sent by Agents for Amazon Bedrock into your Lambda function contains a number of event fields that might be interesting. The event handler exposes them in the `app.current_event` field:
```

Comment:
> might be interesting

Provide an use case.

Reply:
As I wrote above, there's no known use case that I can currently find on public docs. 

> ~[ ] Accessing custom request fields --> make it more meaningful~ -- Couldn't find any relevant documentation or meaningful examples at the moment, so we might add this later

Should we remove this section instead?